File: server_v2/src/db/connection.py
import os
import sqlite3

from src.config import DATABASE_PATH
from src.logging import get_logger

logger = get_logger(__name__)


def get_connection(connection_path: str = DATABASE_PATH):
    logger.debug("Connecting to database", {"connection_path": connection_path})
    folder_name = os.path.dirname(connection_path)
    if not os.path.exists(folder_name):
        error = f"No folder found matching {folder_name}"
        logger.error(error)
        raise ValueError(error)
    connection = sqlite3.connect(connection_path)
    def dict_factory(cursor, row):
        return {col[0]: row[idx] for idx, col in enumerate(cursor.description)}
    connection.row_factory = dict_factory
    return connection

File: server-golang/db/connection.go
// db/connection.go
package db

import (
	"database/sql"
	"os"
	"path/filepath"

	_ "github.com/mattn/go-sqlite3"
)

// Assume DATABASE_PATH from config, like Python.
var DATABASE_PATH = "data/database.db"

func GetConnection(connectionPath string) (*sql.DB, error) {
	if connectionPath == "" {
		connectionPath = DATABASE_PATH
	}
	folder := filepath.Dir(connectionPath)
	if _, err := os.Stat(folder); os.IsNotExist(err) {
		return nil, err
	}
	return sql.Open("sqlite3", connectionPath)
}

// Logging: Use std log for now; replace with custom logger if needed.
